### PR TITLE
SEAB-5116: Miscellaneous notebook tweaks

### PR DESCRIPTION
**Pull request by svonworl**
**Description**
In support of https://ucsc-cgl.atlassian.net/browse/SEAB-5116, this PR makes a couple of small notebook-related changes to the webservice:

1. Corrects a value (ipynb -> jupyter) of the workflow descriptor type.
2. Changes the `ToolDescriptor` class so it imports the TRS openapi `DescriptorType` (rather than defaulting to the version corresponding to the TRS swagger), which causes the new type enum "jupyter" value to be included in the generated webservice openapi and swagger schemas.

This PR changes the "galaxy" type enum value that appears in the corresponding webservice schema files, because it differs between the TRS openapi and swagger files used to build.  Will that break anything?  My initial conclusion is "probably not", but please scrutinize.

**Review Instructions**
Confirm that the "jupyter" value now appears in the `ToolDescriptor` type enum in `openapi.yaml`.

**Issue**
https://ucsc-cgl.atlassian.net/browse/SEAB-5116

**Security**
If there are any concerns that require extra attention from the security team, highlight them here.

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [x] If you are changing dependencies, check the Snyk status check or the dashboard to ensure you are not introducing new high/critical vulnerabilities
- [x] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [x] Do not serve user-uploaded binary images through the Dockstore API
- [x] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [x] Do not create cookies, although this may change in the future
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

### CODE REVIEW

**Review by coverbeck — COMMENTED** [review]
> This PR changes the "galaxy" type enum value that appears in the corresponding webservice schema files, because it differs between the TRS openapi and swagger files used to build. Will that break anything? My initial conclusion is "probably not", but please scrutinize.

This scares me; hopefully you're right, but I see several references to GXFORMAT2 in the UI code. Can you try building the UI against your changes and see if it compiles?

There's also a chance it could break the Galaxy integration. We might need to run it by them.

On the other hand, just looking, this now matches the [GA4GH TRS](https://github.com/ga4gh/tool-registry-service-schemas/blob/develop/openapi/openapi.yaml#L731), so it's probably best to figure it out.

File: dockstore-webservice/src/main/resources/openapi3/openapi.yaml [hunk 1]
```diff
@@ -9720,12 +9720,13 @@ components:
         type:
           type: string
           enum:
-          - SMK
```
Comment (denis-yuen) [inline dockstore-webservice/src/main/resources/openapi3/openapi.yaml:9723]: Question: isn't this an auto-generated on build file? I'm a little confused how this occurred without a corresponding Java code change.

**Review by denis-yuen — COMMENTED** [review]
Agreed with previous comment and one question about how this change occurred. 

Although we're not hardcore about it, we generally avoid making breaking changes since the next Dockstore version is 1.14 (as opposed to 2.0) from a semantic versioning point of view. That said, we can add and deprecate

**Comment by svonworl** [comment]
> > This PR changes the "galaxy" type enum value that appears in the corresponding webservice schema files, because it differs between the TRS openapi and swagger files used to build. Will that break anything? My initial conclusion is "probably not", but please scrutinize.
> 
> This scares me; hopefully you're right, but I see several references to GXFORMAT2 in the UI code. Can you try building the UI against your changes and see if it compiles?
> 
> There's also a chance it could break the Galaxy integration. We might need to run it by them.
> 
> On the other hand, just looking, this now matches the [GA4GH TRS](https://github.com/ga4gh/tool-registry-service-schemas/blob/develop/openapi/openapi.yaml#L731), so it's probably best to figure it out.

So, I did some more digging.  What follows is the motivation for my change, and explanation why it's [probably ok]:

The current values of `ToolDescriptor.TypeEnum` are fine, except that the type and its values are used extensively throughout the ui code (to appreciate how deeply they are woven, do a grep) and the `jupyter` value is missing. There's probably a historical reason for this.

The key takeaway is that, IMHO, it's _much_ easier to adjust the enum values than to change our ui code.

So, `ToolDescriptor.TypeEnum` is derived from `DescriptorType`, which is defined in the TRS schema.  We generate different swagger and openapi versions of `DescriptorType.java` from schemas corresponding to different TRS releases:
https://raw.githubusercontent.com/dockstore/tool-registry-service-schemas/2.0.0-beta.6-smk-gxformat2/src/main/resources/swagger/ga4gh-tool-discovery.yaml
https://raw.githubusercontent.com/dockstore/tool-registry-service-schemas/2.0.7-beta.1-notebooks/openapi/openapi.yaml
These and the associated build steps are in the webservice pom.

So, the values of DescriptorType differ between the two.  The swagger has `GXFORMAT2` and no notebooks value, the openapi has `GALAXY` and `JUPYTER`...

`ToolDescriptor` is a source file that's part of our repo, defined in the `io.swagger.model` package,so it "implicitly imports" the swagger `DescriptorType.java`.  Thus, when the dockstore `openapi.yaml` is generated, it uses the TRS swagger `DescriptorType` values.

So, I changed `ToolDescriptor` to import the openapi `DescriptorType`, which picks up the new notebook value.

The only use of `ToolDescriptor` in the webservice appears to be as part of the response of a few (legacy type?) endpoints.  In fact, the `type` field within can currently only be `CWL` or `WDL`.  See the `ToolDescriptor` constructor.  To confirm that theory, I commented out all other methods in `ToolDescriptor` that might be used to set the `type` field to something else, and the build succeeded.

So, from all this, I tentatively conclude that:
1. my change doesn't alter the responses of the webservice
2. the `ToolDescriptor.TypeEnum` values in the openapi schema don't matter much, as long as the CWL and WDL values remain there, because those are the only two values that webservice will return 
3. except to our UI code, which we'll need to adjust since the `GXFORMAT2` enum value goes away.  I plan to make the adjustment in the "notebook entry page" PR.

That was a lot, thanks for reading.

**Comment by coverbeck** [comment]
Regarding my concerns about the Galaxy integration, I looked into it and the query parameter is already `GALAXY` (they call this endpoint: `https://qa.dockstore.org/api/ga4gh/trs/v2/tools?descriptorType=GALAXY&limit=100`, so it should be good from that perspective.

This is good, it should also allow us to remove some special case UI code we have.

**Comment by denis-yuen** [comment]
> the ToolDescriptor.TypeEnum values in the openapi schema don't matter much, as long as the CWL and WDL values remain there, because those are the only two values that webservice will return

The good thing at a quick scan is that `ToolDescriptor` from a TRS perspective seems to be used only by TRS v1 `dockstore-webservice/src/main/java/io/swagger/api/ToolsApiV1.java`
It looks like TRS v2 beta and final use `FileWrapper` 

In other words, Galaxy may be safe anyway since this could only affect the "wrong" TRS versions, would still like to test to be sure.

**Review by coverbeck — APPROVED** [review]

**Comment by svonworl** [comment]
https://ucsc-cgl.atlassian.net/browse/SEAB-5116

**Pull request merged**